client/packages/system/src/Report/api/hooks/useReportStore.ts

```diff
@@ -0,0 +1,20 @@
+import { create } from '@openmsupply-client/common';
+import { JsonData } from '@openmsupply-client/programs';
+
+interface ReportStoreState {
+  id: string;
+  setId: (reportId: string) => void;
+  name: string;
+  setName: (name: string) => void;
+  args: JsonData | undefined;
+  setArgs: (args: JsonData | undefined) => void;
+}
+
+export const useReportStore = create<ReportStoreState>(set => ({
```

Comment:
Decided to save report information in state rather than create a new endpoint to get the information in the detail view...

Reply:
Will need reports for the update filter modal 👀